Packs/MicrosoftGraphDeviceManagement/Integrations/MicrosoftGraphDeviceManagement/MicrosoftGraphDeviceManagement.py

```diff
@@ -0,0 +1,392 @@
+    }
+}
+
+
+''' COMMANDS '''
+
+
+def list_managed_devices_command(client: MsGraphClient, args: dict) -> Tuple[str, dict, dict]:
+    url_suffix: str = '/deviceManagement/managedDevices'
+    limit: int = try_parse_integer(args.get('limit', 10), err_msg='This value for limit must be an integer.')
+    raw_response = client.make_request('GET', url_suffix)
+    list_raw_devices: list = raw_response.get('value', [])[:limit]
+    list_devices: list = [build_device_object(device) for device in list_raw_devices if device]
+    list_devices_hr: list = [build_device_human_readable(device) for device in list_raw_devices if device]
+    entry_context: dict = {'MSGraphDeviceManagement.Device(val.ID === obj.ID)': list_devices}
+    human_readable: str = 'No managed devices found.'
+    if list_devices:
+        human_readable = tableToMarkdown('List managed devices', list_devices_hr, headers=HEADERS['device'])
+    return human_readable, entry_context, raw_response
+
+
+def get_managed_device_command(client: MsGraphClient, args: dict) -> Tuple[str, dict, dict]:
+    device_id = args.get("device-id")
+    url_suffix: str = f'/deviceManagement/managedDevices/{device_id}'
+    raw_response = client.make_request('GET', url_suffix)
+    device: dict = build_device_object(raw_response)
+    device_hr: dict = build_device_human_readable(raw_response)
+    entry_context: dict = {'MSGraphDeviceManagement.Device(val.ID === obj.ID)': device}
+    device_name: str = device.get('Name', '')
+    human_readable: str = f'Managed device {device_id} not found.'
+    if device:
+        human_readable = tableToMarkdown(f'Managed device {device_name}', device_hr, headers=HEADERS['device'])
+    return human_readable, entry_context, raw_response
+
+
+def make_action_command(client: MsGraphClient, args: dict) -> Tuple[str, dict, dict]:
+    action: str = get_action(demisto.command())
+    body: Union[str, None] = build_request_body(args, action)
+    url_suffix: str = f'deviceManagement/managedDevices/{args.get("device-id")}/{dash_to_camelcase(action)}'
+    client.make_request('POST', url_suffix, data=body, resp_type='None')
+    return f'Device {action.replace("-", " ")} action activated successfully.', {}, {}
+
+
+''' MAIN '''
+
+
+def main():
+    args: dict = demisto.args()
+    params: dict = demisto.params()
+    self_deployed: bool = params.get('self_deployed', False)
+    tenant_id: str = params.get('tenant_id', '')
+    auth_and_token_url: str = params.get('auth_id', '')
+    enc_key: str = params.get('enc_key', '')
+    # remove trailing slash to prevent wrong URL path to service
+    url: str = params.get('url', '')
+    server: str = url[:-1] if (url and url.endswith('/')) else url
+    # service base URL
+    base_url: str = server + '/v1.0'
+    app_name: str = 'ms-graph-device-management'
+    ok_codes: tuple = (200, 201, 202, 204)
+    use_ssl: bool = not params.get('insecure', False)
+    proxy: dict = handle_proxy()
+
+    if self_deployed:
```

Comment:
not sure why we didnt implement it in the module, lets move this condition to the module and make sure graphmail and singleuser support it